src/main/java/org/alfresco/repo/content/ContentStore.java

```diff
@@ -235,4 +235,27 @@
      *      if an IO error occurs
      */
     public boolean delete(String contentUrl);
+
+    /**
+     * Gets a presigned URL to directly access a binary content.
+     *
+     * @param contentUrl A content store URL
+     * @return A direct access URL for a binary content
+     * @throws UnsupportedOperationException if the store is unable to provide the information
+     */
+    default String getDirectAccessUrl(String contentUrl)
```

Comment:
Should we consider adding a second parameter here for the expiration time of the "direct access url"? Should this expiration time be configurable or known only by the content store? @skopf , do you have any suggestions?

Reply:
I'd suggest it was configurable for now via global.properties, we can enhance the feature in the future to allow callers of the API to specify an expiry time/length.

Reply:
To clarify further my suggestion further, the configuration property should also be connector specific.

Reply:
Sounds good, thank you.